### PR TITLE
MINOR: Fix the leak  "unknown" `group.coordinator.rebalance.protocols` on document

### PR DESCRIPTION
**Pull request by m1a2st**
Although "unknown" `group.coordinator.rebalance.protocols` type is a valid string in internal kafka system, but the common user should not config this "unknown" type, so we should remove it from the document.

test in my local
CleanShot 2024-11-16 at 17 39 33

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorConfig.java [hunk 1]
```diff
@@ -76,7 +76,9 @@ public class GroupCoordinatorConfig {
 
     public static final String GROUP_COORDINATOR_REBALANCE_PROTOCOLS_CONFIG = "group.coordinator.rebalance.protocols";
     public static final String GROUP_COORDINATOR_REBALANCE_PROTOCOLS_DOC = "The list of enabled rebalance protocols. Supported protocols: " +
-            Arrays.stream(Group.GroupType.values()).map(Group.GroupType::toString).collect(Collectors.joining(",")) + ". " +
+            Arrays.stream(Group.GroupType.values())
+                    .filter(protocol -> !protocol.equals(Group.GroupType.UNKNOWN))
+                    .map(Group.GroupType::toString).collect(Collectors.joining(", ")) + ". " +
```
Comment (dajac) [inline group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorConfig.java]: I wonder whether we should just drop this part from the documentation given that the values are provided automatically. What do you think?

File: group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorConfig.java [hunk 2]
```diff
@@ -213,7 +215,8 @@ public class GroupCoordinatorConfig {
 
     public static final ConfigDef NEW_GROUP_CONFIG_DEF =  new ConfigDef()
             .define(GROUP_COORDINATOR_REBALANCE_PROTOCOLS_CONFIG, LIST, GROUP_COORDINATOR_REBALANCE_PROTOCOLS_DEFAULT,
-                    ConfigDef.ValidList.in(Utils.enumOptions(Group.GroupType.class)), MEDIUM, GROUP_COORDINATOR_REBALANCE_PROTOCOLS_DOC)
+                    ConfigDef.ValidList.in(Arrays.stream(Utils.enumOptions(Group.GroupType.class))
+                            .filter(groupType -> !groupType.equals(Group.GroupType.UNKNOWN.toString())).toArray(String[]::new)), MEDIUM, GROUP_COORDINATOR_REBALANCE_PROTOCOLS_DOC)
```
Comment (dajac) [inline group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorConfig.java]: nit: I wonder whether we should have a small helper method in `GroupType` to get the list of valid values. What do you think?

**Review by dajac — COMMENTED** [review]
@m1a2st Thanks for the patch. I left a few small suggestions.

File: group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorConfig.java (hunk 1 above)
Comment (m1a2st) [reply group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorConfig.java]: Literally speaking, "unknown" is a recognizable type. But physically, "unknown" is not a supported protocol. In the user documentation, we should hide abnormal settings. WDYT?

Comment (dajac) [reply group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorConfig.java]: Yeah, we should definitely hide unknown. My point was more about removing explicitly mentioning the types in the doc string as they are automatically displayed. We basically show the same list twice in your screenshot.

Comment (chia7712) [reply group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorConfig.java]: @m1a2st we can remove the description "Supported protocols ..." from `GROUP_COORDINATOR_REBALANCE_PROTOCOLS_DOC` as the "Valid Values" has displayed the correct values already.

**Comment by m1a2st** [comment]
test in my local
CleanShot 2024-11-19 at 20 09 34@2x

**Review by chia7712 — APPROVED** [review]

**Review by dajac — APPROVED** [review]
LGTM, thanks.

**Pull request merged**